src/actions/operations/OperationFactory.js

```diff
@@ -0,0 +1,20 @@
+import {Addition} from "./Addition";
+import {Multiplication} from "./Multiplication";
+import {ToSquare} from "./ToSquare";
+
+export class OperationFactory {
+    static allOperations() {
+        return [Addition, Multiplication, ToSquare];
+    }
+
+    static createRandom(category) {
```

Comment:
@MaxiSuppes Creo que se me ocurre una forma más prolija de implementar esto, evitando tener que tener ese método isValidWith, que al menos a mí me suena algo raro. Lo que podes hacer es de, el lugar de pedirle el operator() a la category, pedirle operationClass y que de retorne Addition, Multiplication o ToSquare. Luego a esa clase directamente le pedis createRandom, y sacas este if de "isValidWith".
Me parece mejor tener el if/switch ahí en Category, donde ya hay un acoplamiento en maxSolveTime con las operaciones posibles, que acá.
El "isValidWith" me huele a "isKindOf" o el "instance of" de python, que no están mal, pero si es evitable mejor.

Reply:
@lucianolev Perfecto, y pensaste de que forma obtener la operationClass dentro de un método de category ? De acuerdo al operator de la category debería obtener la operationClass, como te imaginaste este método de category (si es que lo pensaste) ?

Reply:
Sí, un switch sería.